### PR TITLE
Refactor poc_iot_injector to use continuous_source

### PR DESCRIPTION
**Pull request by bbalser**

### CODE REVIEW

File: poc_iot_injector/src/server.rs [hunk 1]
```diff
@@ -48,171 +40,96 @@ impl Server {
...
-            }
             tokio::select! {
                 _ = shutdown.clone() => break,
-                _ = poc_iot_timer.tick() => match self.handle_poc_tick().await {
-                    Ok(()) => (),
-                    Err(err) => {
-                        tracing::error!("fatal poc_iot_injector error: {err:?}");
-                        return Err(err)
-                    }
+                msg = receiver.recv() => if let Some(stream) =  msg {
```
Comment (andymck) [inline poc_iot_injector/src/server.rs:65]: this seems weird in that why would the received msg potentially be a None ? why would the filestore push a None stream

Comment (bbalser) [reply poc_iot_injector/src/server.rs:65]: https://docs.rs/tokio/1.25.0/tokio/sync/mpsc/struct.Receiver.html#method.recv

The `recv` function always returns an `Option`. The file_store is not sending an Option on the channel.

File: poc_iot_injector/migrations/3_files_processed.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+create table files_processed (
+	file_name varchar primary key,
+	file_type varchar,
```
Comment (vihu) [inline poc_iot_injector/migrations/3_files_processed.sql:3]: Do we want to make `file_type`, `file_timestamp` and `processed_at` not null as well?

**Review by vihu — APPROVED** [review]

File: poc_iot_injector/src/server.rs (hunk 1 above)
Comment (maplant) [reply poc_iot_injector/src/server.rs:65]: To further elaborate, recv will return a None if the Sender has been dropped

**Review by maplant — APPROVED** [review]
-98, you love to see it

File: poc_iot_injector/src/server.rs [hunk 3]
```diff
@@ -48,171 +40,96 @@ impl Server {
...
         tracing::info!("starting poc-iot-injector server");
-        let mut poc_iot_timer = time::interval(self.tick_time);
-        poc_iot_timer.set_missed_tick_behavior(time::MissedTickBehavior::Delay);
-
         loop {
-            if shutdown.is_triggered() {
-                break;
-            }
             tokio::select! {
                 _ = shutdown.clone() => break,
```
Comment (jeffgrunewald) [inline poc_iot_injector/src/server.rs:64]: does shutdown still need to be cloned if you're not checking it any longer in the top of the loop?

**Pull request closed without merging**